### PR TITLE
refactor(client): Rewrite `AppMenu` to TypeScript

### PR DESCRIPTION
**Pull request by tiagoevanp**
## Proposed changes (including videos or screenshots)

This is a rewrite of AppMenu from Javascript to Typescript.

**Important note:** there are a lot of possible changes in code enhancement but I chose to deliver the PR as a simple language migration well typed.

## Further comments

[CORE-241](https://rocketchat.atlassian.net/browse/CORE-241)

[CORE-241]: https://rocketchat.atlassian.net/browse/CORE-241?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/meteor/client/components/GenericMenu/GenericMenu.tsx [hunk 1]
```diff
@@ -46,7 +46,7 @@ const GenericMenu = ({ title, icon = 'menu', disabled, onAction, ...props }: Gen
 	const isMenuEmpty = !(sections && sections.length > 0) && !(items && items.length > 0);
 
 	if (isMenuEmpty || disabled) {
-		return <IconButton small icon={icon} disabled />;
+		return <IconButton small icon={icon} title={t.has(title) ? t(title) : title} disabled />;
```
Comment (dougfabris) [inline apps/meteor/client/components/GenericMenu/GenericMenu.tsx:49]: this change isn't necessary at all

File: apps/meteor/client/components/WarningModal.tsx [hunk 2]
```diff
@@ -1,21 +1,21 @@
...
-	cancel: () => void;
-	cancelText: string;
-	confirm: () => Promise<void>;
 };
 
 const WarningModal = ({ text, confirmText, close, cancel, cancelText, confirm, ...props }: WarningModalProps): ReactElement => {
 	const t = useTranslation();
 	return (
-		<Modal {...props}>
+		<Modal open {...props}>
```
Comment (dougfabris) [inline apps/meteor/client/components/WarningModal.tsx:18]: why?

File: apps/meteor/client/views/marketplace/AppMenu.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+	if (isLoading) {
+		return <Skeleton variant='rect' height='x28' width='x28' />;
+	}
+
+	if (!isAdminUser && app?.installed) {
+		return null;
+	}
+
+	return <Menu title={t('More_options')} options={menuOptions} placement='bottom-start' maxHeight='initial' />;
```
Comment (dougfabris) [inline apps/meteor/client/views/marketplace/AppMenu.tsx:26]: why not using `MenuV2`?

File: apps/meteor/client/views/marketplace/hooks/useAppMenu.tsx [hunk 4]
```diff
@@ -8,74 +10,92 @@ import {
...
-
-function AppMenu({ app, isAppDetailsPage, ...props }) {
+import WarningModal from '../../../components/WarningModal';
+import { useIsEnterprise } from '../../../hooks/useIsEnterprise';
+import IframeModal from '../IframeModal';
+import UninstallGrandfatheredAppModal from '../components/UninstallGrandfatheredAppModal/UninstallGrandfatheredAppModal';
+import type { Actions } from '../helpers';
+import { appEnabledStatuses, appButtonProps } from '../helpers';
+import { handleAPIError } from '../helpers/handleAPIError';
+// import { marketplaceActions } from '../helpers/marketplaceActions';
```
Comment (dougfabris) [inline apps/meteor/client/views/marketplace/hooks/useAppMenu.tsx:24]: ?

**Review by dougfabris — CHANGES_REQUESTED** [review]

File: apps/meteor/client/components/GenericMenu/GenericMenu.tsx (hunk 1 above)
Comment (tiagoevanp) [reply apps/meteor/client/components/GenericMenu/GenericMenu.tsx:49]: I think the ternary is not necessary indeed...

Comment (tiagoevanp) [reply apps/meteor/client/components/GenericMenu/GenericMenu.tsx:49]: As discussed previously, disabled buttons don't need a title on it. Removing title prop!

File: apps/meteor/client/components/WarningModal.tsx (hunk 2 above)
Comment (tiagoevanp) [reply apps/meteor/client/components/WarningModal.tsx:18]: Dialog element need the open prop to expose their content (aria-hidden): https://testing-library.com/docs/queries/byrole/#hidden

**Review by ggazzo — APPROVED** [review]

**Pull request merged**